### PR TITLE
Add the ability to retrieve a list of tenants for the tenants endpoint

### PR DESCRIPTION
**Pull request by mattgresham**
- Bump library version
- Add the ability to retrieve a list of tenants (SP-674)

### CODE REVIEW

File: Library/SPIClient.h [hunk 1]
```diff
@@ -597,4 +598,15 @@ suppressMerchantPassword:(BOOL)suppressMerchantPassword
  */
 - (void)getTerminalConfiguration;
 
+/**
+ * Static call to retrieve the available tenants (payment providers) for mx51. This is used to display the payment providers available in your Simple Payments Integration setup.
+ * @param posVendorId NSString
```
Comment (gouline) [inline Library/SPIClient.h]: Fill these in please.

File: Library/SPIClient.m [hunk 2]
```diff
@@ -1170,6 +1171,26 @@ - (void)autoResolveEftposAddress {
     }];
 }
 
+#pragma Static Methods
+
+/**
```
Comment (gouline) [inline Library/SPIClient.m]: You don't need to repeat the comment block in the implementation, header file one is sufficient.

File: Library/Service/SPITenantsService.m [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+@implementation SPITenantsService : NSObject
+
+- (void)retrieveTenants:(NSString *)posVendorId
+                 apiKey:(NSString *)apiKey
+            countryCode:(NSString *)countryCode
+             completion:(SPITenantsResult)completion {
+    NSString *tenantServiceUrl = [NSString stringWithFormat: @"https://spi.integration.mspenv.io/tenants?country-code=%@&pos-vendor-id=%@&api-key=%@", countryCode, posVendorId, apiKey];
+    
+    NSMutableURLRequest *request = [[NSMutableURLRequest alloc] init];
```
Comment (gouline) [inline Library/Service/SPITenantsService.m]: It's bad practice to manually alloc and init built-in classes, use one of the helpers such as `requestWithURL`.

File: Library/Service/SPITenantsService.m [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+        NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse *)urlResponse;
+        if ([httpResponse statusCode] != 200){
+            NSLog(@"Error getting %@, HTTP status code %li", tenantServiceUrl, (long)[httpResponse statusCode]);
+            
+            completion(nil);
+            return;
+        }
+        
+        NSError *JSONError = nil;
```
Comment (gouline) [inline Library/Service/SPITenantsService.m]: Don't start local variables with an uppercase letter, use `jsonError`, `jsonData`, `jsonDictionary`.

**Review by gouline — CHANGES_REQUESTED** [review]
Good work, minor comments.

**Comment by mattgresham** [comment]
> Good work, minor comments.

Thanks for taking a look @gouline, I appreciate the feedback. I've updated the PR, please let me know if there are any problems with my updates?

**Review by gouline — APPROVED** [review]

**Pull request merged**